markets/retrievaladapter/provider.go

```diff
@@ -135,10 +135,14 @@ func (rpn *retrievalProviderNode) GetRetrievalPricingInput(ctx context.Context,
 	}
 	tsk := head.Key()
 
+	var lastErr error
```

Comment:
Let's have this be a multierror, so we can accumulate all errors instead of just the last one.